mod_reforged/hooks/skills/effects/bleeding_effect.nut

```diff
@@ -0,0 +1,131 @@
+	q.m.Stacks <- 1;
+	q.m.StacksThisTurn <- 1;
+	q.m.StacksForMoraleCheck <- 3;
+	q.m.MaxMalusMult <- 0.5;
+	q.m.MalusMultPerStack <- 0.05;
+
+	q.create = @(__original) function()
+	{
+		__original();
+		this.m.Description = "This character is bleeding profusely and will lose hitpoints each turn while acting slower and losing courage.";
+		this.m.IsStacking = false;
+	}
+
+	q.getTooltip <- function()
+	{
+		local ret = this.skill.getTooltip();
+		ret.push({
+			id = 10,
+			type = "text",
+			icon = "ui/icons/regular_damage.png",
+			text = ::Reforged.Mod.Tooltips.parseString("Receive " + ::MSU.Text.colorRed(this.getDamage()) + " damage per [turn|Concept.Turn]")
+		});
+		ret.push({
+			id = 11,
+			type = "text",
+			icon = "ui/icons/bravery.png",
+			text = ::Reforged.Mod.Tooltips.parseString(::MSU.Text.colorizeMult(this.getMalusMult()) + " reduced [Resolve|Concept.Bravery]")
+		});
+		ret.push({
+			id = 12,
+			type = "text",
+			icon = "ui/icons/initiative.png",
+			text = ::Reforged.Mod.Tooltips.parseString(::MSU.Text.colorizeMult(this.getMalusMult()) + " reduced [Initiative|Concept.Initiative]")
+		});
+		ret.push({
+			id = 13,
+			type = "text",
+			icon = "ui/icons/special.png",
+			text = ::Reforged.Mod.Tooltips.parseString("If at least " + ::MSU.Text.colorRed(this.m.StacksForMoraleCheck) + " stacks are received in a single [turn|Concept.Turn], a negative [morale check|Concept.Morale] is immediately triggered")
+		});
+		return ret;
+	}
+
+	q.getName <- function()
+	{
+		return this.m.Stacks == 1 ? this.m.Name : this.m.Name + " (x" + this.m.Stacks + ")";
+	}
+
+	q.getDescription = @() function()
+	{
+		return this.m.Description;
+	}
+
+	q.getMalusMult <- function()
+	{
+		return 1.0 - ::Math.minf(this.m.MaxMalusMult, this.m.Stacks * this.m.MalusMultPerStack);
+	}
+
+	q.getDamage <- function()
+	{
+		local ret = this.getContainer().hasSkill("trait.bleeder") ? this.m.Stacks * 2 : 1;
+		return this.getContainer().hasSkill("effects.hyena_potion") ? ::Math.floor(ret * 0.5) : ret;
+	}
+
+	// The vanilla file has an empty onUpdate function
+	q.onUpdate = @() function( _properties )
+	{
+		local mult = this.getMalusMult();
+		_properties.InitiativeMult *= mult;
+		_properties.BraveryMult *= mult;
+	}
+
+	// Based on the vanilla function but use this.getDamage() to calculate hitInfo.DamageRegular and use this.m.Overlay for spawnIcon
+	// and get rid of the part that decrements this.m.TurnsLeft and removes the effect
+	// also apply damage in instances smaller than ::Const.Morale.OnHitMinDamage to prevent morale check from this damage
+	q.applyDamage = @() function()
+	{
+		if (this.m.LastRoundApplied != ::Time.getRound())
+		{
+			this.m.LastRoundApplied = ::Time.getRound();
+			local actor = this.getContainer().getActor();
+			this.spawnIcon(this.m.Overlay, actor.getTile());
+
+			// Split the damage instances if damage is greater than ::Const.Morale.OnHitMinDamage
```

Comment:
Yeah I considered that morale state thing but didn't like it because of potential breakage with things. I'd rather break Nine Lives than change the morale state which may have many knock-on effects on other skills. But I agree this isn't good. But I don't like the morale state solution either. We could try to brainstorm a better solution.

Reply:
How is it an issue if my switcheroo potentially breaks things in the future, but not now.
But it's not an issue that your implementation potentially breaks things in the future (once someone introduces an effect that increases damage taken).

Meanwhile my solution prevents the nine lives edge case aswell as allowing compartmentalizing of the bleeder trait effect and potentially other effects that increase bleed damage taken.

Reply:
> But it's not an issue that your implementation potentially breaks things in the future but not now.

First off, my solution breaks things (i.e. Nine Lives) **now**, not potentially in the future. Secondly, I didn't say it's not an issue, in fact I said "I agree this isn't good".

Your proposed solution changes the morale state which may have completely "surprise" effects with other perks/skills/effects that we later add or that submodders add because no one expects a human's morale state to be `Ignore` in the middle of damage. So I don't like that solution. It can break things in a very opaque and unexpected way. 

Which is why I said let's try to brainstorm a better solution.